realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1037,23 +1037,27 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFind(
     return -1;
 }
 
-std::unique_ptr<Query> getHandoverQuery (jlong bgSharedGroupPtr, jlong queryPtr, bool advanceToLatestVersion)
+// Returns a pointer to query on the worker SharedGroup or throw a BadVersion if the SharedGroup version required
+// for the handover is no longer available.
+std::unique_ptr<Query> handoverQueryToWorker(jlong bgSharedGroupPtr, jlong queryPtr, bool advanceToLatestVersion)
```

Comment:
Do we still need `advanceToLatestVersion` for https://github.com/realm/realm-java/pull/2294/files#diff-feb202ebcef8ca848c17b2f2f81e78e7R2455 ?